docs/proposal/20250324-rosa-iam-roles.md

```diff
@@ -0,0 +1,226 @@
+# Propose ROSA HCP IAM Account Roles, IAM Operator Roles, and OIDC Configuration as CRD in CAPA
+
+## Table of Contents
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Problem Statement](#problem-statement)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+- [Resources](#resources)
+
+## Summary
+The ROSA HCP in CAPA (Cluster API Provider AWS) project currently lacks built-in support for managing IAM account roles, IAM operator roles, and OIDC configuration as Custom Resource Definitions (CRDs). This proposal suggests integrating the [ROSA library](https://github.com/openshift/rosa) to extend ROSA-HCP's functionality, enabling declarative IAM and OIDC management.
```

Comment:
Would it be worth adding a way to declare IAM account roles within CAPA's shared API? Or does it not make sense in self-managed clusters?

EDIT: looks like this is addressed later under the alternatives section. Restating it here - this proposal's primarily aimed about integrating existing code from ROSA's library into a controller. This proposal isn't concerned with any other type of AWS cluster.

Reply:
yes, its only about ROSA-HCP clusters. The ROSA lib add by default some tags that are required by OCM (hypershift) to identify/own the roles.